### PR TITLE
[FLINK-11577][tests] Replace StackTraceSampleCoordinatorITCase 

### PR DESCRIPTION
**Pull request by GJL**
## What is the purpose of the change

*This removes the StackTraceSampleCoordinatorITCase test, and replaces it with `StackTraceSampleServiceTest#testShouldReturnPartialResultIfTaskStopsRunningDuringSampling`*

## Brief change log

  - *See commits*

## Verifying this change

This change added tests and can be verified as follows:

  - *Added unit test `StackTraceSampleServiceTest`*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/StackTraceSampleServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.lessThan;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+
+/**
+ * Tests for {@link StackTraceSampleService}.
+ */
+public class StackTraceSampleServiceTest {
```
Comment (tillrohrmann) [inline flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/StackTraceSampleServiceTest.java]: Test should extend from `TestLogger`

File: flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/StackTraceSampleServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+		final ScheduledExecutor scheduledExecutor = new ScheduledExecutorServiceAdapter(scheduledExecutorService);
+
+		stackTraceSampleService = new StackTraceSampleService(scheduledExecutor);
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		if (scheduledExecutorService != null) {
+			scheduledExecutorService.shutdownNow();
+			scheduledExecutorService.awaitTermination(10, TimeUnit.SECONDS);
```
Comment (tillrohrmann) [inline flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/StackTraceSampleServiceTest.java]: We could use `ExecutorUtils#gracefulShutdown` here.

File: flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/StackTraceSampleServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+	}
+
+	@Test
+	public void testShouldThrowExceptionIfTaskIsNotRunningBeforeSampling() throws Exception {
+		try {
+			stackTraceSampleService.requestStackTraceSample(
+				new NotRunningTask(),
+				10,
+				Time.milliseconds(0),
+				-1).get();
```
Comment (tillrohrmann) [inline flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/StackTraceSampleServiceTest.java]: `fail` is missing.

File: flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/StackTraceSampleServiceTest.java (hunk 2 above)
Comment (GJL) [reply flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/StackTraceSampleServiceTest.java]: Will fix before merging.

File: flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/StackTraceSampleServiceTest.java (hunk 1 above)
Comment (GJL) [reply flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/StackTraceSampleServiceTest.java]: Will fix before merging.

File: flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/StackTraceSampleServiceTest.java (hunk 3 above)
Comment (GJL) [reply flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/StackTraceSampleServiceTest.java]: Will fix before merging.

**Comment by GJL** [comment]
Thanks for the review. Will merge as soon as build is green.

**Pull request merged**